hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientInvocation.java

```diff
@@ -171,6 +172,7 @@ public void notify(ClientMessage clientMessage) {
         clientInvocationFuture.complete(clientMessage);
     }
 
+    @SuppressWarnings({ "checkstyle:cyclomaticcomplexity", "checkstyle:npathcomplexity" })
```

Comment:
I am not in favor of pulling out methods when methods are not reused. But I have a pr cookinf up built on top of this. It will simplify these parts by introducing invocation classes for different targets. There I will remove back the suppressions.